### ISSUE TITLE
Off-by-one error with message splitting causes "Message too long" kick

### ISSUE THREAD

**Issue by RoyCurtis** [rocketman10404/tabbychat#22]
If a message is 101 characters long, TabbyChat still regards it as "1" message instead of "2". However, the Minecraft chat message limit is 100 characters.
## Reproduction

_Requires connection to a multiplayer server_
1. Copy and paste the following into chat:

> /msg VanDerProtofsky this is a very looooooong message testttttttttttttttttttttttttttttttttttttttttt
## Expected

TabbyChat splits the above 101 character long message into two messages, with the second message being a single "t"
## Actual

TabbyChat sends the above 101 character long message as a single message, causing the server to kick for "Message Too Long"

**Comment by RoyCurtis** [comment]
This may be a duplicate of issue #21, however there is not enough information in that issue report for me to decide so

**Comment by rocketman10404** [comment]
FWIW, in this specific case, I'd just like to be clear that the expected result would be the following two messages:
- Msg 1 `this is a very looooooong message`
- Msg 2 `testttttttttttttttttttttttttttttttttttttttttt`

as TabbyChat will only split messages at word boundaries.  In fact, that could cause a problem if someone every types > 100 characters with no spaces.  In that case, TabbyChat should attempt to send the entire word as one message, likely resulting in a kick.

Regardless, your example should not result in a server-kick, so I'll look into it further.

**Comment by rocketman10404** [comment]
Yes, I'm seeing a problem I referenced while closing #18.  TabbyChat screws up when a message hits exactly 100 characters.  If you double-check, you'll see the example you provided is, in fact, exactly 100 characters.  Delete one `t` from the end, and it works just fine - add one `t` to the end, and TabbyChat splits it into two messages, which both work just fine.

```
/msg VanDerProtofsky this is a very looooooong message testttttttttttttttttttttttttttttttttttttttttt
1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
         1         2         3         4         5         6         7         8         9         0
                                                                                                   1
```

The root cause of the problem is how TabbyChat splits a message into works, by spaces, and then re-assembles them - checking the length to make sure the next word won't push the message over the limit.  Specifically, the problem occurs because TabbyChat adds the whitespace 'glue' to the end of each word, and trims it off when sending all but the last message (which could also be the only message).

In this case, you've provided 100 characters, which SHOULD actually work.  TabbyChat knows this, and attempts to send it all in one message.  However, since that's also the last message, TabbyChat adds an extra space on the end prior to sending, pushing the message over the actual 100-character limit, causing the player to get kicked.

Thanks for the report, this should be fixed in the next release.

**Comment by rocketman10404** [comment]
Fixed in 0350faafbeea4deb985f9925390aae5f04b3d5fc

**Comment by RoyCurtis** [comment]
Ah ok; I most likely mistook "columns" for characters in Sublime Text. Glad this still lead to a resolution, thank you for the fix!
